mantidproject/mantid
# Handle infinite sized components in instrument viewer

Description of work.
This PR adds a check that the current component has finite length. If it does not it sets that component to always hidden and will not consider it. 

It also contains a fix where the viewport of the instrument view also accounted for hidden components which could affect scale. 

This is only a temporary fix. An issue has been opened here #19684 to address this at IDF level. 

**To test:**
- Run `LoadEmptyInsrument` with the *WISH_Definition_10Panels.xml* IDF
- Open the instrument view on the workspace.
- Set the render mode (top left) to *Full 3D*
- You should see the full instrument and the scale should be appropriate
- Click the instrument tab at the top of the interface
- Click a panel in the tree, all other components should be hidden with only that panel displayed

Fixes #19539 .

Release Notes [here](https://github.com/mantidproject/mantid/pull/19683/files#diff-5528c2656dbf0459917bd0b5ae7b0539)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
